### PR TITLE
Move parsing functionality to `ncbi.model.NCBITaxonomy`

### PR DESCRIPTION
**Pull request by sygao**
Deserialized record data now handled within `NCBITaxonomy`, rather than within `NCBIClient` methods.

Brings `NCBITaxonomy` functionality into line with `NCBINuccore`.

### CODE REVIEW

File: virtool_cli/ncbi/model.py [hunk 1]
```diff
@@ -63,7 +63,36 @@ class NCBILineage(BaseModel):
...
+
+    @field_validator("lineage", mode="before")
+    @classmethod
+    def create_lineage_list(cls, raw: list[dict]):
+        lineage = []
+        for level_data in raw:
+            level = NCBILineage(**level_data)
+            lineage.append(level)
+
+        return lineage
```
Comment (igboyes) [inline virtool_cli/ncbi/model.py:93]: Use list comprehension.

**Review by igboyes — CHANGES_REQUESTED** [review]
* Conflicts with `alpha` need to be resolved before merge
* Look at Deepsource issues.
* Add opening comment for PR so I know what it is doing. I think I get it and seems good to me.

**Comment by sygao** [comment]
> * Look at Deepsource issues.

We might want to exclude `PYL-R0201` from test files. I don't think that recommended fix plays well with pytest.

**Comment by igboyes** [comment]
I think [this PR](https://github.com/virtool/virtool-cli/pull/125) should resolve the problem. We can just ignore in tests.

**Pull request merged**